matrix/basic_matrix.py

```diff
@@ -33,10 +33,12 @@ def __getitem__(self, key):
             return self.ls_entries[key]
         if len(key) == 2:
             row, col = key
-            if all(isinstance(i, int) for i in key):
+            if all(isinstance(i, int) for i in key) or isinstance(col, slice):
                 return self.ls_entries[row][col]
-            elif any(isinstance(i, slice) for i in key):
```

Comment:
If the column is a slice, the return should actually be self.ls_entries[row][col]